shuup/admin/modules/orders/static_src/create/view/lines.js

```diff
@@ -134,8 +134,56 @@ export function renderOrderLines(store, shop, lines) {
     }).compact().value();
 }
 
+var select2 = {
+    clear: function() {
+        if (select2.element) {
+            select2.element.select2("val", null);
+        }
+    },
+    view: function(ctrl, attrs) {
+        return m("select", {
+            config: select2.config(attrs), "data-model": "shuup.product"});
+    },
+    config: function(ctrl) {
+        return function(element, isInitialized) {
+            if(typeof jQuery !== "undefined" && typeof jQuery.fn.select2 !== "undefined") {
+                var el = $(element);
+                select2.element = el;
+                if (!isInitialized) {
+                    // This is needed to reset prop when going back from confirmation view
+                    productQuickSelect.currentProduct(null);
+                    el.select2()
+                        .on("change", function() {
+                            ctrl.onchange($(this).val());
+                        });
+                }
+            } else {
+                alert(gettext("Missing JavaScript dependencies detected"));
+            }
+        };
+    }
+};
+
+var productQuickSelect = {
+    clearSelection: function() {
+        productQuickSelect.currentProduct(null);
+        select2.clear();
+    },
+    currentProduct: m.prop(),
+    changeProduct: function(id) {
+        productQuickSelect.currentProduct(id);
+    },
+    view: function() {
+        return m.component(select2, {
+            onchange: this.changeProduct,
+        });
+    }
+};
+
 export function orderLinesView(store, isCreating) {
     const {lines, shop} = store.getState();
+    // This is needed to make select work when going back from confirmation view
+    store.dispatch(updateTotals(store.getState));
```

Comment:
Not sure exactly why this is needed, but somehow the select2 element isn't initializing correctly without it when pressing "Back" from confirmation view. Kind of hacky, so if anyone has a better idea let me know :smile: